### DISCUSSION TITLE
Accessors ignore eager loads

### DISCUSSION THREAD

**Discussion by PeraMika** [laravel/framework#33831] [General] [answered]
- Laravel Version: 6.18.19
- PHP Version: 7.3.11
- Database Driver & Version: MariaDB 10.4.8

### Description:

Accessors ignore eager loads (and execute queries separately). 

### Steps To Reproduce:

For example, I have the following 1-to-1 relationship between `User` and `Membership` models:  

- `User` *belongsTo* `Membership`

```
// User.php:

    public function membership()
    {
        return $this->belongsTo(Membership::class);
    }
```

On `User` model I have the following accessor:

```
    public function getSignatureAttribute()
    {
        if ($this->membership->is_premium && $this->membership->allows_signature) {
            return $this->membership->signature;
        } else {
            return $this->last_name;
        }
    }
``` 

The problem is when you want to display a listing of users:

```
// In controller:
$users = User::where('verified', 1)
              ->with('membership')
              ->paginate(20);

// In blade:
@foreach($users as $user)
        <div>{{ $user->signature }}</div>
@endforeach
```
This will execute as many queries as there are `$users` because the `signature` accessor ignores eager-load (`->with('membership')`). 
So, for every `$user->signature` it will execute one query to get `membership`.

**Comment by staudenmeir** [comment, marked as answer]
This should definitely work.

What does the query log look like? Can you reproduce the issue on a fresh Laravel installation?

**Comment by PeraMika** [comment]
My bad, sorry. I got confused with other things not being eagerly loaded (and was thinking that's because of accessor ignoring eager-loads). Sorry!
